### PR TITLE
Extension Module list on ReadtheDocs

### PR DESCRIPTION
**Pull request by laceysanderson**
# Documentation
Issue #402 

## Description
This is meant to replace the extension modules list on tripal.info with the hopes of being easier to maintain and contribute to. Additionally it has the benefit of directly linking modules to the Tripal version they are compatible with and giving extension module developers commit credit in Tripal via documentation :-)

## Testing?
Build the docs (`make html` in the dcs directory. See http://www.sphinx-doc.org/en/master/usage/installation.html if you dont have Sphinx installed yet). 

**New documentation available at "Extension Modules"**: check for correct descriptions and missing modules.

**Alternatively you can look at the documentation here: https://tripal.readthedocs.io/en/402_rtd_module_list/extensions.html**

## Screenshots (if appropriate):
screen shot 2018-11-16 at 12 33 05 pm

## Questions to ask yourself
- Are these good categories for extension modules?
- Would it be easy to contribute to and maintain?
- Is it easy to find a module you know exists?
- Are any Tripal3-compatible modules missing?

### CODE REVIEW

**Comment by bradfordcondon** [comment]
Some thoughts:

>Make sure to follow alphabetical order when choosing where on the category page to add your module.

If we used a TOC and loaded all modules in the folder, thsi would happen automatically.  Downside, each module would need to be its own file.

>Include links to both the documentation (even if it’s your README) and the repository (e.g. Github, Gitlab)

can we clarify that in order for the PR to be accepted, both links are **required**

relatedly, its kind of interesting.  we totally steamroll the drupal module distribution option via drupal, drush, etc).  i have some of the blame here because i dont like it, but we should maybe allow for it?


I would also say we should define some standard of "what is a tripal 3 module" that makes it elligible for this list.

File: docs/extensions.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+==================
+
+The below modules are Tripal 3 compatible and grouped roughly by category:
+
+.. toctree::
+   :maxdepth: 2
+
+   extensions/administrative
+   extensions/analysis
+   extensions/datainput
```
Comment (bradfordcondon) [inline docs/extensions.rst]: in the rest of the guide, we use snake case.  let's do so here as well for consistency.

File: docs/extensions/developer.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+Developer Tools
+===============
+
+The following modules provide support to both core and extension Tripal developers.
+
+Tripal Docker (Centos)
```
Comment (bradfordcondon) [inline docs/extensions/developer.rst]: tehse are deprecated.  i made an issue about it in the module.  I'm going to replace it with our tripaldock tool.

**Review by bradfordcondon — APPROVED** [review]

**Comment by spficklin** [comment]
First off. This is Fantastic!  Lots of work went into this, so thanks!

> Are these good categories for extension modules?

I like these categories.  I don't think I would change any.

> Would it be easy to contribute to and maintain?

Yes. I think anyone who develops modules could easily edit the doc and provide a pull request.

> Is it easy to find a module you know exists?

I'm not sure it matters. If you know the name of the module I think a Google search is where you'd go first.  So, I don't think that's as a big a need as advertising to new users what's out there.

> Are any Tripal3-compatible modules missing?

I'd like to add my Tripal Network module.  But that raises a question.  What modules do we list?  Do we require a full release of a module or component before they are listed?  We don't want to send folks to a module that is still in development and not likely to perform as expected.  Maybe we could have a section titled 'Modules in Development'?  And add a comment above them that developers are open to collaboration on these modules?

edit: I just re-read @bradfordcondon 's post above and saw he brought up the same question....

**Comment by spficklin** [comment]
Oh the Tripal Map module is missing https://tripal.info/extensions/modules/tripalmap

**Comment by spficklin** [comment]
@bradfordcondon wrote:
> relatedly, its kind of interesting. we totally steamroll the drupal module distribution option via drupal, drush, etc). i have some of the blame here because i dont like it, but we should maybe allow for it?

Yes, I think we should allow for it.  But we should encourage folks to put their modules up on Drupal.org as full projects so that the updating is more smooth.

**Comment by bradfordcondon** [comment]
An in development section is a great idea. 👍

**Comment by bradfordcondon** [comment]
I added an in development section... but do we have any modules to put there?  My Tripal Apollo module, maybe.  TripalMap until its t3 compatible.  Anything else?

**Comment by spficklin** [comment]
I'm adding the On Hold label so that we can sort out or rating scheme before we push this. If you disagree just say so.

**Review by spficklin — APPROVED** [review]

**Comment by spficklin** [comment]
I took the "On Hold" off of this and merged.  I was thinking about this again and thought it was silly to hold it while we waited on issue #774 .  My approval adds to @bradfordcondon so I've merged.

**Pull request merged**